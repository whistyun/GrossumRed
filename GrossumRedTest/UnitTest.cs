using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using GrossumRed;
using NUnit.Framework;
using System.IO;

namespace GrossumRedTest
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest
    {
        static UnitTest()
        {
            Approvals.RegisterDefaultNamerCreation(() => new Renamer("Out"));
        }

        [Test]
        public void Test1()
        {
            var source = Utils.LoadText("VmCls.vm.cs");
            var generated = CompileManager.CompileString(source, "test", errInfo => { });

            Approvals.Verify(generated);
        }

        [Test]
        public void Test2()
        {
            var source = Utils.LoadText("VmCls2.vm.cs");
            var generated = CompileManager.CompileString(source, "test", errInfo => { });

            Approvals.Verify(generated);
        }
    }


    class Renamer : UnitTestFrameworkNamer
    {
        private string dir;

        public override string SourcePath
        {
            get => Path.Combine(base.SourcePath, dir);
        }

        public Renamer(string dir)
        {
            this.dir = dir;
        }
    }
}