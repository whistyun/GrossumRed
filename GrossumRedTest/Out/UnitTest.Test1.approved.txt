#line 1 "test"
using System;
#line default

#line 3 "test"
namespace Hoge
{
#line default

#line 5 "test"
    public class VmCls : System.Object
    
#line default
,System.ComponentModel.INotifyPropertyChanged{
public event System.ComponentModel.PropertyChangedEventHandler PropertyChanged;

#line 7 "test"
        public int PropertyA {
#line default

get{ return __PropertyA; }

set {
    if(Object.ReferenceEquals(__PropertyA, value)) return;
    
    if(Object.ReferenceEquals(__PropertyA, null) || !__PropertyA.Equals(value)) 
    {
        __PropertyA = value;
        
        if(PropertyChanged!=null)
        {
            PropertyChanged.Invoke(
                this,
                new System.ComponentModel.PropertyChangedEventArgs("PropertyA"));


        }
    }
}

#line 7 "test"
 }
#line default
private int  __PropertyA ;

#line 8 "test"
        public string PropertyB {
#line default

get{ return __PropertyB; }

set {
    if(Object.ReferenceEquals(__PropertyB, value)) return;
    
    if(Object.ReferenceEquals(__PropertyB, null) || !__PropertyB.Equals(value)) 
    {
        __PropertyB = value;
        
        if(PropertyChanged!=null)
        {
            PropertyChanged.Invoke(
                this,
                new System.ComponentModel.PropertyChangedEventArgs("PropertyB"));


        }
    }
}

#line 8 "test"
 }
#line default
private string  __PropertyB ;

#line 10 "test"
        class InnerVmCls
        
#line default
:System.ComponentModel.INotifyPropertyChanged{
public event System.ComponentModel.PropertyChangedEventHandler PropertyChanged;

#line 12 "test"
            public List<string> PropertyB {
#line default

get{ return __PropertyB; }

set {
    if(Object.ReferenceEquals(__PropertyB, value)) return;
    
    if(Object.ReferenceEquals(__PropertyB, null) || !__PropertyB.Equals(value)) 
    {
        __PropertyB = value;
        
        if(PropertyChanged!=null)
        {
            PropertyChanged.Invoke(
                this,
                new System.ComponentModel.PropertyChangedEventArgs("PropertyB"));


        }
    }
}

#line 12 "test"
 }
#line default
private List<string>  __PropertyB ;

#line 13 "test"
        }
#line default

#line 14 "test"
    }
#line default

#line 15 "test"
}
#line default
